builder/azure/arm/step_deploy_template.go

```diff
@@ -58,6 +58,23 @@ func (s *StepDeployTemplate) deployTemplate(ctx context.Context, resourceGroupNa
 	return err
 }
 
+func (s *StepDeployTemplate) deleteTemplate(ctx context.Context, state multistep.StateBag) error {
+	var resourceGroupName = state.Get(constants.ArmResourceGroupName).(string)
```

Comment:
Is ArmResourceGroupName _always_ set? Otherwise I think the assertion could fail.

Reply:
Looks like the answer is yes -- it's either a temp name generated by Packer or a custom name given by the user.